src/main/java/org/openstreetmap/atlas/checks/event/CheckFlagEvent.java

```diff
@@ -54,27 +63,82 @@ public static JsonObject flagToFeature(final CheckFlag flag,
+                    .getAsJsonArray(GEOMETRIES);
+            geometriesWithProperties.stream().forEach(geometry -> Optional
+                    .ofNullable(geometry.getProperties()).ifPresent(propertyMap ->
+                    {
+                        final JsonObject properties = new JsonObject();
+                        propertyMap.forEach(
+                                (key, value) -> properties.addProperty(key, (String) value));
+                        featureProperties.add(properties);
+
+                        Optional.ofNullable(properties.get("osmid")).ifPresent(featureOsmIds::add);
+                    }));
         }
         else
         {
-            feature = GEOJSON_BUILDER.createGeometryCollection(geometries).jsonObject();
+            feature = new JsonObject();
+            geometriesJsonArray = new JsonArray();
+        }
+        if (!flaggedRelations.isEmpty())
+        {
+            final JsonArray geometriesOfFlaggedRelations = new JsonArray();
+            if (!feature.has(TYPE))
+            {
+                feature.addProperty(TYPE, "Feature");
+            }
+            // Get flagged relations as GeoJson features
+            final List<JsonObject> flaggedRelationFeatures = flaggedRelations.stream()
+                    .map(flaggedRelation -> flaggedRelation.asGeoJsonFeature(flag.getIdentifier()))
+                    .collect(Collectors.toList());
+            if (flaggedRelations.size() == 1 && !feature.has(GEOMETRY))
+            {
+                feature.add(GEOMETRY, flaggedRelationFeatures.get(0).get(GEOMETRY));
+            }
+            else if (flaggedRelations.size() != 1 && !feature.has(GEOMETRY))
+            {
+                final JsonObject geometryCollection = new JsonObject();
+                geometryCollection.add(GEOMETRIES, geometriesJsonArray);
+                geometryCollection.addProperty(TYPE, GEOMETRY_COLLECTION);
+                feature.add(GEOMETRY, geometryCollection);
+            }
+
+            // Add all geometries of flaggedRelations to the json array
+            flaggedRelationFeatures.stream()
+                    .map(flaggedRelationFeature -> flaggedRelationFeature.get(GEOMETRY))
+                    .forEach(jsonElement -> geometriesOfFlaggedRelations.add(jsonElement));
+            // To geometries of flagged objects add geometries of flaggedRelation
+            geometriesJsonArray.addAll(geometriesOfFlaggedRelations);
+            // Update feature properties from flaggedRelation features
+            flaggedRelationFeatures.stream()
+                    .map(flaggedRelationFeature -> flaggedRelationFeature.get(PROPERTIES))
+                    .forEach(property -> featureProperties.add(property));
+            // Add osm id
+            flaggedRelationFeatures.stream()
+                    .map(flaggedRelationFeature -> flaggedRelationFeature.get(PROPERTIES))
+                    .map(jsonElement -> jsonElement.getAsJsonObject().get("osmIdentifier"))
```

Comment:
This was originally `osmid`, I'd suggest sticking with that

Reply:
The asGeoJsonFeature() populates the osmid property as "osmIdentifier" (in GeoJsonUtils). I am fetching that id to update the `feature_osmids`.

Reply:
Ahh, okay nice